pipeline/pipelines/batch_generation_pipeline.py

```diff
@@ -0,0 +1,133 @@
+# limitations under the License.
+
+"""Pipelines that run GAPIC"""
+
+import yaml
+import glob
+
+from pipeline.pipelines import code_generation_pipeline as code_gen
+from pipeline.tasks import batch_tasks
+from taskflow.patterns import linear_flow
+from pipeline.utils import config_util, pipeline_util, task_utils
+
+
+class BatchPipeline(code_gen.CodeGenerationPipelineBase):
+
+    def __init__(self, get_pipeline_tasks_func, **kwargs):
+        task_factory = BatchTaskFactory(get_pipeline_tasks_func)
+        super(BatchPipeline, self).__init__(
+            task_factory, **kwargs)
+
+    def validate_kwargs(self, **kwargs):
+        required = ['apis', 'languages', 'api_config_pattern',
+                    'artman_language_yaml', 'batch_output_dir']
+        pipeline_util.validate_exists(required, **kwargs)
+
+
+class BatchTaskFactory(code_gen.TaskFactoryBase):
+
+    def __init__(self, get_pipeline_tasks_func):
+        self.get_pipeline_tasks_func = get_pipeline_tasks_func
+        super(BatchTaskFactory, self).__init__()
+
+    def get_tasks(self, **kwargs):
+        # TODO(michaelbausor): convert to unordered flow if necessary
+        batch_flow = linear_flow.Flow('BatchFlow')
+        for single_flow in self.get_language_api_flows(**kwargs):
+            batch_flow.add(single_flow)
+
+        # TODO(michaelbausor): add task to create Github PR once staging repo
+        # is public on Github
+        final_tasks = []
+
+        flow = linear_flow.Flow('BatchGenerationFlow')
+        flow.add(batch_flow)
+        flow.add(*task_utils.instantiate_tasks(final_tasks, kwargs))
+
+        return flow
+
+    def get_language_api_flows(self, apis, languages, api_config_pattern,
+                               artman_language_yaml, **kwargs):
+        # Remove language arg from kwargs - we don't want it to override
+        # the language set in _get_configs
+        del kwargs['language']
+
+        repo_root = kwargs['repo_root']
+        api_config_yamls = _get_api_configs(api_config_pattern, apis)
+
+        for api_kwargs in _get_configs(
+                api_config_yamls, languages, artman_language_yaml, repo_root):
+
+            api_kwargs.update(kwargs)
+
+            tasks = self.get_pipeline_tasks_func(**api_kwargs)
+            tasks += task_utils.instantiate_tasks(
+                [batch_tasks.BatchOutputDirTask,
+                 batch_tasks.BatchCleanTask,
+                 batch_tasks.BatchCopyTask],
+                api_kwargs)
+
+            single_flow = linear_flow.Flow('SingleLanguageApiFlow')
+            single_flow.add(*tasks)
+            yield single_flow
+
+
+def _get_configs(api_config_yamls, languages, artman_language_yaml, repo_root):
+    for language in _get_languages(artman_language_yaml, languages):
+        lang_config = _load_artman_config(artman_language_yaml,
+                                          language,
+                                          repo_root)
+        lang_config['language'] = language
```

Comment:
There is not a strong reason to do it here rather than lower down, just that we know the language at this point and don't need to set it for each API in the inner loop. I can move it down to api_kwargs